services/requester/requester.go

```diff
@@ -131,17 +131,23 @@ func NewEVM(
 		signer: signer,
 		logger: logger,
 		blocks: blocks,
+		txPool: NewTxPool(client, logger),
 	}
 
 	// create COA on the account
 	if config.CreateCOAResource {
 		// we ignore errors for now since creation of already existing COA resource will fail, which is fine for now
-		id, err := evm.signAndSend(
+		tx, err := evm.buildTransaction(
 			context.Background(),
 			evm.replaceAddresses(createCOAScript),
 			cadence.UFix64(coaFundingBalance),
 		)
-		logger.Warn().Err(err).Str("id", id.String()).Msg("COA resource auto-creation status")
+		if err != nil {
+			logger.Warn().Err(err).Msg("COA resource auto-creation failure")
+		}
+		if err := evm.client.SendTransaction(context.Background(), *tx); err != nil {
+			logger.Warn().Err(err).Msg("COA resource auto-creation failure")
```

Comment:
```suggestion
			logger.Warn().Err(err).Msg("error while send transaction to the network")
```

Reply:
I want to keep context in which transaction failed: COA creation, because otherwise we won't know in logs what failed. I can update the message to be more clear it was sending COA transaction tho.